src/rgw/rgw_rest.cc

```diff
@@ -1636,6 +1636,14 @@ int RGWHandler_REST::validate_bucket_name(const string& bucket)
     return -ERR_INVALID_BUCKET_NAME;
   }
 
+  const char *s = bucket.c_str();
+  for (int i = 0; i < len; ++i, ++s) {
+    if (*(unsigned char *)s == 0xff)
+      return -ERR_INVALID_BUCKET_NAME;
+    if (*(unsigned char *)s == '/')
+      return -ERR_INVALID_BUCKET_NAME;
+  }
+
```

Comment:
Why not use cpp style as 
```cpp
for (auto c : bucket) {
    if ( c == 0xff ||
         c == '/')
       return -ERR_INVALID_BUCKET_NAME;
}
```

Reply:
We can probably drop the whole C++ -> C string conversion and just do a string.find() or probably even using std::any_of